File: src/main/java/com/online/store/service/OrdersService.java
package com.online.store.service;

import com.online.store.entity.Orders;

import java.util.List;

/**
 * Service interface for Orders.class
 *
 * @author Dmitri Tsornoi
 */
public interface OrdersService {

    List<Orders> findAll();

    Orders findOne(Long id);

    void save(Orders order);

    void update(Orders orders);

    void delete(Long id);

    void restore(Long id);

    List<Orders> findByUserAccountId(Long id);
}

File: src/main/frontend/online-store/src/app/service/orders.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

/**
 * Service for Orders retrieveing and presisting in backend
 *
 * @author Mark Salumaa
 */


@Injectable({
  providedIn: 'root'
})
export class OrdersService {

  private URL = '/api/orders';

  constructor(private http: HttpClient) {
  }

  getAll(): Observable<any>{
    return this.http.get(`${this.URL}/`);
  }

  get(id): Observable<any>{
    return this.http.get(`${this.URL}/${id}`);
  }

  getbyUserId(id): Observable<any>{
    return this.http.get(`${this.URL}/user/${id}`);
  }

  create(order): Observable<any>{
    return this.http.post(`${this.URL}/`, order);
  }

  update(order): Observable<any>{
    return this.http.put(`${this.URL}/`, order);
  }

  delete(order): Observable<any>{
    return this.http.post(`${this.URL}/delete`, order);
  }

  restore(order): Observable<any>{
    return this.http.post(`${this.URL}/restore`, order);
  }

}
